csv.h

```diff
@@ -462,9 +491,14 @@ namespace io{
                                 }
                         }
 
+						bool quoted_data = false;
                         int line_end = data_begin;
-                        while(buffer[line_end] != '\n' && line_end != data_end){
+                        while( (quoted_data || buffer[line_end] != '\n') && line_end != data_end){
```

Comment:
when using double quote policy, new_line() should not stop prematurely on the first \n it sees but rather on the first un-quoted \n.